lib/js/jsre.nim

```diff
@@ -3,27 +3,30 @@
 when not defined(js):
   {.error: "This module only works on the JavaScript platform".}
 
-type RegExp* {.importjs.} = ref object of JsRoot
+type RegExp* = ref object of JsRoot
   ## Regular Expressions for JavaScript target.
   ## See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
-  flags* {.importjs.}: cstring        ## cstring that contains the flags of the RegExp object.
-  dotAll* {.importjs.}: bool          ## Whether `.` matches newlines or not.
-  global* {.importjs.}: bool          ## Whether to test against all possible matches in a string, or only against the first.
-  ignoreCase* {.importjs.}: bool      ## Whether to ignore case while attempting a match in a string.
-  multiline* {.importjs.}: bool       ## Whether to search in strings across multiple lines.
-  source* {.importjs.}: cstring       ## The text of the pattern.
-  sticky* {.importjs.}: bool          ## Whether the search is sticky.
-  unicode* {.importjs.}: bool         ## Whether Unicode features are enabled.
-  lastIndex* {.importjs.}: cint       ## Index at which to start the next match (read/write property).
-  input* {.importjs.}: cstring        ## Read-only and modified on successful match.
-  lastMatch* {.importjs.}: cstring    ## Ditto.
-  lastParen* {.importjs.}: cstring    ## Ditto.
-  leftContext* {.importjs.}: cstring  ## Ditto.
-  rightContext* {.importjs.}: cstring ## Ditto.
+  flags*: cstring        ## cstring that contains the flags of the RegExp object.
+  dotAll*: bool          ## Whether `.` matches newlines or not.
+  global*: bool          ## Whether to test against all possible matches in a string, or only against the first.
+  ignoreCase*: bool      ## Whether to ignore case while attempting a match in a string.
+  multiline*: bool       ## Whether to search in strings across multiple lines.
+  source*: cstring       ## The text of the pattern.
+  sticky*: bool          ## Whether the search is sticky.
+  unicode*: bool         ## Whether Unicode features are enabled.
+  lastIndex*: cint       ## Index at which to start the next match (read/write property).
+  input*: cstring        ## Read-only and modified on successful match.
+  lastMatch*: cstring    ## Ditto.
+  lastParen*: cstring    ## Ditto.
+  leftContext*: cstring  ## Ditto.
+  rightContext*: cstring ## Ditto.
+
 
 func newRegExp*(pattern: cstring; flags: cstring): RegExp {.importjs: "new RegExp(@)".}
   ## Creates a new RegExp object.
 
+func newRegExp*(pattern: cstring): RegExp {.importjs: "new RegExp(@)".}
+
```

Comment:
now that i realize that `test` is a substring match, how about naming it contains instead of match, and changing the order of arguments:

rationale:
* in pkg/regex, https://nitely.github.io/nim-regex/regex.html#match%2Cstring%2CRegex%2CRegexMatch%2Cint uses `contains` for this: `func contains(s: string; pattern: Regex): bool`
* in pkg/regex, `match` indicates a full match, and `find` is for a substring match (but also takes extra params: `func find(s: string; pattern: Regex; m: var RegexMatch; start = 0): bool`
* ditto with std/re: https://nim-lang.github.io/Nim/re.html#contains%2Cstring%2CRegex%2Cint
* ditto with std/nre: https://nim-lang.github.io/Nim/nre.html#contains%2Cstring%2CRegex%2Cint

basically, remove `match` and add this:
```nim
func contains*(pattern: cstring, self: RegExp): bool =
  ## Tests for a substring match in its string parameter.
  runnableExamples:
    let r = newRegExp(r"bc$", r"i")
    assert r in r"abc"
    assert r notin r"abcd"
    assert "xabc".contains(r)
  asm "`result` = `self`.test(`pattern`)"
```